tests/SharedTests/TestLeads.cs

```diff
@@ -0,0 +1,620 @@
+        {
+            var lead = new Lead
+            {
+                FirstName = "Test",
+                LastName = "Lead"
+            };
+            var currency = new TransactionCurrency()
+            {
+                ExchangeRate = 0.5m,
+                CurrencyPrecision = 2
+            };
+
+            lead.Id = orgAdminUIService.Create(lead);
+            currency.Id = orgAdminUIService.Create(currency);
+
+            var request = new QualifyLeadRequest
+            {
+                Status = new OptionSetValue((int)Lead_StatusCode.Qualified),
+                LeadId = lead.ToEntityReference(),
+                OpportunityCurrencyId = currency.ToEntityReference()
+            };
+
+            var response = orgAdminUIService.Execute(request) as QualifyLeadResponse;
+            Assert.IsNotNull(response);
+            var createdEntities = response.CreatedEntities;
+            Assert.IsNotNull(createdEntities);
+            Assert.IsTrue(createdEntities.Count == 0);
+        }
+
+        [TestMethod]
+        public void TestQualifyLeadRequestOpportunityCustomerIdSuccess()
+        {
+            var lead = new Lead
+            {
+                FirstName = "Test",
+                LastName = "Lead"
+            };
+            var account = new Account();
+            lead.Id = orgAdminUIService.Create(lead);
+            account.Id = orgAdminUIService.Create(account);
+
+            var request = new QualifyLeadRequest
+            {
+                Status = new OptionSetValue((int)Lead_StatusCode.Qualified),
+                LeadId = lead.ToEntityReference(),
+                OpportunityCustomerId = account.ToEntityReference()
+            };
+
+            var response = orgAdminUIService.Execute(request) as QualifyLeadResponse;
+            Assert.IsNotNull(response);
+            var createdEntities = response.CreatedEntities;
+            Assert.IsNotNull(createdEntities);
+            Assert.IsTrue(createdEntities.Count == 0);
+        }
+
+        [TestMethod]
+        public void TestQualifyLeadRequestCreateAccountSuccess()
+        {
+            var lead = new Lead
+            {
+                FirstName = "Test",
+                LastName = "Lead"
+            };
+            lead.Id = orgAdminUIService.Create(lead);
+
+            var request = new QualifyLeadRequest
+            {
+                Status = new OptionSetValue((int)Lead_StatusCode.Qualified),
+                LeadId = lead.ToEntityReference(),
+                CreateAccount = true
+            };
+
+            var response = orgAdminUIService.Execute(request) as QualifyLeadResponse;
+            Assert.IsNotNull(response);
+            var createdEntities = response.CreatedEntities;
+            Assert.IsNotNull(createdEntities);
+            Assert.IsTrue(createdEntities.Count == 1);
+        }
+
+        [TestMethod]
+        public void TestQualifyLeadRequestCreateContactSuccess()
```

Comment:
Missing assert that lead has been qualified and check if a new contact have been created